### PR TITLE
Threading fixes

### PR DESCRIPTION
**Pull request by PatrickvL**
These changes are an attempt to fix issue #692 (TLS Issue for Certain Titles.)

Could someone do a regressions test with this, please?

* EmuGenerateFS : Still initialize when TLS data copy isn't needed
* Introduced InitXboxThread for consistent Xbox thread initialization
* Rename EmuInitFS into PatchFSOpcodes
* Threading fixes …Keep EmuUpdateTickCount on Xbox core, EmuThreadDpcHandler requires Xbox thread setup too, removed commented PCSTProxy thread creation.

PS: Excuse my commit-history, it's from before I started rebasing.

### CODE REVIEW

File: src/CxbxKrnl/EmuD3D8.cpp [hunk 1]
```diff
@@ -395,7 +395,7 @@ const char *CxbxGetErrorDescription(HRESULT hResult)
 	case DDERR_NOTLOCKED: return "An attempt was made to unlock a surface that was not locked.";
 	case DDERR_NOTPAGELOCKED: return "An attempt was made to page-unlock a surface with no outstanding page locks.";
 	case DDERR_NOTPALETTIZED: return "The surface being used is not a palette-based surface.";
-	case DDERR_NOVSYNCHW: return "There is no hardware support for vertical blank�synchronized operations.";
+	case DDERR_NOVSYNCHW: return "There is no hardware support for vertical blanksynchronized operations.";
```
Comment (PatrickvL) [inline src/CxbxKrnl/EmuD3D8.cpp]: Unintentional none-change, please ignore.

**Comment by LukeUsher** [comment]
I'm not sure about the rename of EmuInitFS to PatchFsOpcodes, the logic being that it is that patching is not the only task it performs: What happens once we have a functioning JIT, FS instructions are no longer patched, but EmuInitFS still needs to be called to setup the KPCR structures.

Also EmuUpdateTickCount should not be kept on the Xbox core: It may not be called often enough for accurate timing in this case.

File: src/CxbxKrnl/EmuD3D8.cpp [hunk 2]
```diff
@@ -2298,7 +2293,7 @@ HRESULT WINAPI XTL::EMUPATCH(Direct3D_CreateDevice)
 		//g_pD3DDevice8->SetTextureStageState(stage, D3DTSS_COLORKEYOP, D3DTCOLORKEYOP_DISABLE);
 		//g_pD3DDevice8->SetTextureStageState(stage, D3DTSS_COLORSIGN, 0);
 		//g_pD3DDevice8->SetTextureStageState(stage, D3DTSS_ALPHAKILL, D3DTALPHAKILL_DISABLE);
-		g_pD3DDevice8->SetTextureStageState(stage, D3DTSS_COLOROP, D3DTOP_MODULATE); //  D3DTOP_DISABLE caused textures to go opaque
+		g_pD3DDevice8->SetTextureStageState(stage, D3DTSS_COLOROP, D3DTOP_SELECTARG1); //  D3DTOP_DISABLE caused textures to go opaque
```
Comment (PatrickvL) [inline src/CxbxKrnl/EmuD3D8.cpp]: Don't know where this came from - could be the result of an incorrect conflict resolve. Leave it out if possible.

**Comment by RadWolfie** [comment]
I am advising not to merge this pull request for the following reasons:
- Too many merges in this pull request
- Is not sync'd/rebased to latest master branch.
- Has other non-relative commits to the pull request tasks.

If I was the owner of this repo, I will not accept this pull request for the reasons above.

What I do advise is...
- Create new pull request after branch is sync'd to master.
- Add commits relative to the issue only.

**Comment by LukeUsher** [comment]
The merge commits are a valid concern, it will be addressed by either Patrick or myself later

**Comment by PatrickvL** [comment]
The actual changes are small enough to warrant a squash and merge - shall I do that instead?

**Comment by PatrickvL** [comment]
Oh, and would it be possible for someone to do a regression-test with this, please?

**Comment by literalmente-game** [comment]
Games tested:
- Smashing Drive
- Burnout
- Gauntlet Dark Legacy
- Dolphin (Tech Demo)
- Arx Fatalis
- Battlestar Galactica
- Unreal Championship
- Taz - Wanted
- ZSNES BOX (SNES emulator)

Regressions/differences on behavior:

- Arx Fatalis: Dirty disc warning on booting
- Unreal Championship: now shows an error message box with no text and two buttons
- Taz Wanted: the background image on the loading screen is now messed up:
taz
- ZSNES BOX (SNES emulator): now crash with an error after showing the main menu:
znes

**Comment by PatrickvL** [comment]
Wow, those effects I didn't expect!

**Comment by literalmente-game** [comment]
I've tested the same games for the last PR Luke did and we had no changes in behavior compared to the last master.

Arx Fatalis: previously loaded a black screen, with music/sounds running on the background and then display an error message box (music will continue to play unless you click "Ok"/"Cancel"). Now I got the dirty disc warning directly on booting, no sound at all (so I don't know if it's better for displaying something... or bad for not playing the music as before).

Unreal Championship: on the master builds, we can see the logos and the game crash before the menu. On this PR, the empty error/warning message box is shown (with two buttons that have no text on them). Selecting any button will just crash the emulator (so, crashing as before, but for a different reason)

Taz Wanted: on the master builds we can see the pre rendered videos, then the loading screen with a loading bar + background image (the image is shown as intended). On this PR, everything is the same, except for the background image that now is shown as in the image I've posted.

ZSNES BOX: on the master you can boot, load a game and run it with no problems (besides some timing issues, making the games run at least at 4x the real speed). On this PR: crashes after displaying the main menu, showing that error message on the image I've posted

**Comment by PatrickvL** [comment]
Your latest compatibility issue seems to indicate the regressions are independent of this pr?

**Comment by BenNottelling** [comment]
@literalmente-game ZSNES Box didn't regress with this PR

**Comment by PatrickvL** [comment]
It'd be good if reports always specify commit hashes that were tested.

**Comment by literalmente-game** [comment]
@BenNottelling It did, I've a clean HLE cache. I can reproduce this every single time. I've deleted all the files on the "Partition1","Partition2","Partition3","Partition4", etc. folders, just to be sure. Could it be system specific?

@PatrickvL on 8617b2a9 ZSNES works, on 363685b8 it shows the error shown on my post

**Comment by LukeUsher** [comment]
The ZSNES regression is a division by zero error within Decode@CWMADecoder@@QAGJPBU_XMEDIAPACKET@@@Z

**Comment by LukeUsher** [comment]
Update: I was able to reproduce the ZSNES regression in master too, it's not specific to this branch afterall.
I cleared the HLE Cache, and it still breaks, the regression must be elsewhere.

**Comment by ergo720** [comment]
I also add that Constantine has regressed too, I already submitted a report for that in the compatibility repository

**Comment by LukeUsher** [comment]
The regressions above were unrelated to this PR,

**Comment by PatrickvL** [comment]
Thanks for merging this, but you should have squashed it - now Radwolfie will have my head

**Comment by PatrickvL** [comment]
@literalmente-game Could you check if any of the regressions you mentioned above still exist, and create separate issues per regression? *Please* mention commit hashes from working and not-working builds, to help pin-point root causes.

**Pull request merged**